### PR TITLE
Feature: UI page

### PR DESCRIPTION
**Pull request by yuzhouliu9**
image

UI:
- Distribution dropdowns get populated from GET `/api/v1beta1/distributions` with the user's token cookie
- On clicking Submit, will create POST request to `/api/v1beta1/distributions/{distribution}/invalidations` or GET request to `/api/v1beta1/distributions/{distribution}/invalidations/{id}`
- Result is displayed in Output box as JSON from requests

Minor changes to server code to host static files.

Deployed to test cluster.
Can try with distribution: `sandbox-infra-test`. Valid path prefix is `/infra-test-only/`

### CODE REVIEW

File: internal/core/v1beta1/distribution_service.go [hunk 1]
```diff
@@ -56,6 +57,7 @@ func (d *DistributionService) List(ctx context.Context) ([]string, error) {
 	for name := range distributions {
 		ret = append(ret, name)
 	}
+	sort.Strings(ret)
```
Comment (yuzhouliu9) [inline internal/core/v1beta1/distribution_service.go:60]: this is to give consistent distribution list response in alphabetical order.

File: internal/cli/cli.go [hunk 2]
```diff
@@ -89,6 +89,7 @@ func (c *RootCommand) runE(cmd *cobra.Command, args []string) error {
 	s, err := server.New(
 		config,
 		cloudfrontClient,
+		server.WithAuthCookieName(viper.GetString("auth-cookie")),
```
Comment (yuzhouliu9) [inline internal/cli/cli.go:92]: I missed setting this option before.

File: internal/server/ui/app.js [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+
+async function getInvalidation() {
+    // show loading spinner
+    document.getElementById("loading").style.visibility="visible";
+
+    // construct the GET request
+    let distribution = document.getElementById("get-invalidation-distribution").value;
+    let invalidationID = document.getElementById("get-invalidation-id").value;
+    invalidationID = invalidationID.replace(/^"(.*)"$/, "$1");  // remove surrounding quotes if any
```
Comment (dskatz) [inline internal/server/ui/app.js:144]: Add a test to error out if invalidationID is empty.

File: internal/server/ui/app.js [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    // send POST request
+    await postJson(url, { paths: pathsArr })
+    .then(data => {
+        appendOutput("Create", details, data);
+    })
+    .catch(error => {
+        appendOutput("Create Error", details, error.message);
+    });
+
+    // hide loading spinner
```
Comment (dskatz) [inline internal/server/ui/app.js:131]: Clear the paths form field.

File: internal/server/ui/app.js [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+async function createInvalidation() {
+    // show loading spinner
+    document.getElementById("loading").style.visibility="visible";
+
+    // construct the POST request
+    let distribution = document.getElementById("create-invalidation-distribution").value;
+    let url = apiPrefix.concat("/", distribution, "/invalidations");
+
+    let commaSeparatedPaths = document.getElementById("create-invalidation-paths").value;
+    let pathsArr = commaSeparatedPaths.split(",").map(function(item) {
```
Comment (dskatz) [inline internal/server/ui/app.js:115]: If pathsArr is empty, error out before making the fetch.

File: internal/server/ui/index.html [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+  <!-- Required meta tags -->
+  <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  
+  <!-- Bootstrap CSS -->
+  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/css/bootstrap.min.css" integrity="sha384-zCbKRCUGaJDkqS1kPbPd7TveP5iyJE0EjAuZQTgFLD2ylzuqKfdKlfG/eSrtxUkn" crossorigin="anonymous">
+
+  <style>
+    /* CSS for loading spinner */
+    #loading {width: 100%;height: 100%;top: 0px;left: 0px;position: fixed;display: block; z-index: 99}
```
Comment (dskatz) [inline internal/server/ui/index.html:13]: Add a partially transparent "greyed" out background to make the spinner more apparent.

**Review by dskatz — COMMENTED** [review]
Just a few minor improvements, overall looks good.

As discussed in slack:

- reversing the order of the output, most recent on top
- adding a timestamp to the event of the output. e.g. `Create : 00:00PM`

**Comment by redondos** [comment]
> * reversing the order of the output, most recent on top
> * adding a timestamp to the event of the output. e.g. `Create : 00:00PM`

It would also be useful to give focus to the most recent entry in the output log on submission, whether it is an invalidation request or invalidation status.

**Review by dskatz — APPROVED** [review]
